scripts/zones/Western_Altepa_Desert/mobs/King_Vinegarroon.lua

```diff
@@ -49,23 +49,22 @@ entity.onMobWeaponSkill = function(target, mob, skill)
 
     -- Every time KV performs a TP move, he will draw in either his target or the entire alliance randomly
     if (skill:getID() == 354 or skill:getID() == 355 or skill:getID() == 722 or skill:getID() == 723) and os.time() > drawInWait then
-        local chance = math.random(1,2)
+        local chance = math.random(1, 2)
         if chance == 1 then
             mob:triggerDrawIn(mob, true, 1, 35, target)
         else
             mob:triggerDrawIn(mob, false, 1, 35, target)
         end
 
         -- KV always does an AOE TP move followed by a single target TP move
-        mob:useMobAbility(({ 353,350,720 })[math.random(1,3)])
+        mob:useMobAbility(({ 353, 350, 720 })[math.random(1, 3)])
         mob:setLocalVar("DrawInWait", os.time() + 2)
     end
 end
 
 entity.onMobDespawn = function(mob)
-    UpdateNMSpawnPoint(mob:getID())
     local respawn = 75600 -- 21h
-    mob:setRespawnTime(respawn)
+    xi.mob.nmTODPersist(mob, respawn)
     mob:setLocalVar("respawn", os.time() + respawn)
```

Comment:
This LocalVar should be changed to a server variable or his respawn. If the server crashes or the zone goes down its being reset anyway. I think you can utilize the server variable set in side the nmTODPersist and just replace the one in the weatherchange function in the zone lua